tests/unit/test_graph_parser.py

```diff
@@ -133,13 +133,33 @@ def test_graph_syntax_errors_2(seq, graph, expected_err):
             "foo || bar => baz",
             "The graph OR operator is '|'"
         ),
+        param(
+            # See https://github.com/cylc/cylc-flow/issues/5844
+            "foo => bar[1649]",
+            'illegal cycle point offset on the right',
+            id='no-cycle-point-RHS'
+        ),
+        param(
+            "foo => bar[1649] => baz",
+            '!illegal cycle point offset on the right',
+            id='ignore-not-exclusively-RHS-cycle-point'
+        ),
+        param(
+            "foo => bar[1649]\nbar[1649] => baz",
+            '!illegal cycle point offset on the right',
+            id='ignore-not-exclusively-RHS-cycle-point2'
+        ),
```

Comment:
Better suited to including in `test_trigger_equivalence` instead?

Reply:
I'm not sure that I see why.

Reply:
Because the test is about errors and there are no errors in these two examples

Reply:
I'll move it if you insist, but I think that checking errors aren't raised when they shouldn't be is a valid error test.

Reply:
A test that doesn't assert that an exception is raised *is* testing that errors aren't raised

Reply:
It's simpler to have two tests, one for the success and one for the error cases, rather than trying to merge the two:

```python
@pytest.mark.parametrize(...)
def test_do_something_good(foo):
    do_something(foo)

@pytest.mark.parametrize(...)
def test_do_something_bad(foo):
    with pytest.raises(...):
        do_something(foo)
```

Nothing wrong with having multiple tests (note that's what parametrize is doing anyway).

I like to think of test cases in terms of the "good", the "bad" and the "ugly", i.e. expected success cases, expected error cases & nasty edge cases designed to test the extremes. This wild west approach sometimes leaks into test names :grin:.